tests/test_build_changes.py

```diff
@@ -32,6 +46,45 @@ def test_build(app):
         ' Can now be replaced with a different allocator.</a>')
     assert malloc_html in htmltext
 
+    assert '[new config] 4 added' in app._status.getvalue()
+
+    # config changed (due to how the fixture works, changed master_doc)
+    args, kwargs = app_params
+    app0 = make_app(*args, **kwargs)
+    assert app0.env.config_status == CONFIG_CHANGED
+    app0.build()
+    assert "[config changed ('master_doc')] 4 added" in app0._status.getvalue()
+
+    # no changes
+    confoverrides = dict(master_doc=app0.env.config['master_doc'])
+    app1 = make_app(*args, confoverrides=confoverrides, **kwargs)
+    assert app0.env.config_status == CONFIG_OK
+    app1.build()
+    assert "0 added, 0 changed, 0 removed" in app1._status.getvalue()
+
+    # change a file
+    fname = os.path.join(app.srcdir, confoverrides['master_doc'] + '.rst')
+    with open(fname, 'a') as fid:
+        fid.write('\nfoo')
+    app2 = make_app(*args, confoverrides=confoverrides, **kwargs)
+    assert app2.env.config_status == CONFIG_OK
+    app2.build()
+    assert "0 added, 1 changed, 0 removed" in app2._status.getvalue()
+
+    # extensions changed (this check does not trigger properly...?)
```

Comment:
@tk0miya if you look I think these changes should trigger a CONFIG_EXTENSIONS_CHANGED but they trigger a CONFIG_NEW. You can see what I tried below. Instead of this `conf.py` writing approach, I also tried:
```
monkeypatch.setattr(sphinx.ext.todo, 'setup', lambda app: {'version': '0.00'})
```
Which seemed like it should work (detect changed version of extension) but it did not.

I'm not sure if this is a problem with the test fixtures, Sphinx, or my usage of them. But in any case the code is almost fully tested now, just the EXTENSIONS_CHANGED case is missing, which might be good enough for this PR.

Reply:
Good point. It seems there is a bug. I'll try to fix it.

Reply:
I just posted https://github.com/sphinx-doc/sphinx/pull/6532. It will raise `CONFIG_EXTENSIONS_CHANGED` status.